Library/Homebrew/cmd/update.rb

```diff
@@ -398,10 +398,26 @@ def select_formula(key)
 
   def dump_formula_report(key, title)
     formula = select_formula(key)
+    formula_installed = select_installed_formula(key, formula)
     formula.map! { |oldname, newname| "#{oldname} -> #{newname}" } if key == :R
     unless formula.empty?
       ohai title
-      puts_columns formula
+      puts_columns(formula, formula_installed)
     end
   end
+
+  def select_installed_formula(key, formula)
+    return formula.select { |name| installed?(name) } unless key == :R
+
+    formula.
+      select { |oldname, newname| installed?(oldname) || installed?(newname) }.
+      map { |oldname, newname| "#{oldname} -> #{newname}" }
+  end
+
+  def installed?(formula)
+    f = Formulary.factory(formula)
+    f.installed?
+  rescue
+    false
```

Comment:
Yes, seems wasteful to load the entire formula for such a simple check. So something like `(HOMEBREW_CELLAR / formula.split("/", 3).last).directory?` to also get it right for taps?